### PR TITLE
Implement Generic Applications references migration in pages

### PR DESCRIPTION
**Pull request by boubaker**
A new Data Updgrade plugin is added by this comit to allow replace an application reference inside pages by a new one. a concrete example is added here to replace old Calendar application references to new Agenda add application references

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -24,14 +24,16 @@
...
     <!-- 3rd party libraries versions -->
-    <org.exoplatform.social.version>6.1.x-SNAPSHOT</org.exoplatform.social.version>
-    <addon.exo.ecms.version>6.1.x-SNAPSHOT</addon.exo.ecms.version>
-    <addon.exo.wiki.version>6.1.x-SNAPSHOT</addon.exo.wiki.version>
-    <addon.exo.jcr.version>6.1.x-SNAPSHOT</addon.exo.jcr.version>
+    <org.exoplatform.social.version>6.1.x-agenda-SNAPSHOT</org.exoplatform.social.version>
+    <addon.exo.ecms.version>6.1.x-agenda-SNAPSHOT</addon.exo.ecms.version>
+    <org.exoplatform.social.version>6.1.x-agenda-SNAPSHOT</org.exoplatform.social.version>
+    <addon.exo.wiki.version>6.1.x-agenda-SNAPSHOT</addon.exo.wiki.version>
+    <addon.exo.jcr.version>6.1.x-agenda-SNAPSHOT</addon.exo.jcr.version>
```
Comment (hbenali) [inline pom.xml:36]: IMO, these versions will not be available as we removed their related CI Jobs

File: pom.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
   </parent>
   <groupId>org.exoplatform.addons.upgrade</groupId>
   <artifactId>upgrade</artifactId>
-  <version>6.1.x-SNAPSHOT</version>
+  <version>6.1.x-agenda-SNAPSHOT</version>
```
Comment (hbenali) [inline pom.xml:13]: I think we can keep that version inside a FB, WDYT ?

File: pom.xml (hunk 1 above)
Comment (boubaker) [reply pom.xml:36]: Yep, true, I will change this, thanks @hbenali

File: pom.xml (hunk 2 above)
Comment (boubaker) [reply pom.xml:13]: It's already in FB in one commit of Omar:
https://github.com/exo-addons/data-upgrade/pull/11/commits/545d8d4170815aef02c3bc61dd225752e2fff079
in this PR you will find two commits, once will be pushed to develop and another to generate a dedicated version of the addon, no more.

File: pom.xml (hunk 1 above)
Comment (boubaker) [reply pom.xml]: Done

**Pull request closed without merging**